### ISSUE TITLE
Support for Cabal's (eventual) show-build-info

### ISSUE THREAD

**Issue by robinp** [commercialhaskell/stack#2139]
Opening to track plumbing through the support for Cabal's show-build-info (https://github.com/haskell/cabal/pull/2771) in stack.

This would be useful, since then third party tooling could easily execute GHC API based commands on the project.

@bgamari @DanielG

**Comment by mgsloan** [comment]
Good idea!  Would be helpful for having ghci support the configuration modifications from Setup.hs  https://github.com/commercialhaskell/stack/issues/970#issuecomment-157612875

Does it also provide a way to run `initialBuildSteps`? See https://github.com/commercialhaskell/stack/issues/1364
